### PR TITLE
Update list and addressDuplicate endpoints to enable pagination

### PR DESCRIPTION
**Pull request by LBHCallumM**
## Describe this PR

- List endpoint now includes 'pageNumber' parameter, which defaults to 1
- DuplicateAddress endpoint includes 'pagesize' and 'pageNumber' parameters
- The 'pageNumber' parameter starts at 1
- PageSize defaults to 25, as that is the number of results shown on the frontend

- Both endpoints now return `ListGasServiceDataRequest`

```javascript
{
  "pageNumber": 1,
  "totalResults": 24,
  "totalPages": 1,
  "data": [
    ... existingResponse
  ]
}
```

## Required Frontend Changes

- Accept different response objects for gasServiceData and addressDuplicates endpoints
- Handle serverside pagination from datagrid
- When fetching data for PDF's, pass max pageSize parameter (1000), and check for more results if pageNumber is less than totalPages

### CODE REVIEW

File: RepairsApi/V2/Boundary/Request/ListGasServiceDataRequest.cs [hunk 1]
```diff
@@ -62,11 +62,13 @@ public class ListGasServiceDataRequest
...
+        public ReportType? ReportType { get; set; }
+
         [FromQuery(Name = "pagesize")]
         public int PageSize { get; set; } = 100;
 
-
-        [FromQuery(Name = "reporttype")]
-        public ReportType? ReportType { get; set; }
+        [FromQuery(Name = "page")]
+        public int Page { get; set; } = 0;
```
Comment (Asura5) [inline RepairsApi/V2/Boundary/Request/ListGasServiceDataRequest.cs:72]: Any reason why you've gone for a zero based page numbering system?

Comment (LBHCallumM) [reply RepairsApi/V2/Boundary/Request/ListGasServiceDataRequest.cs:72]: It's slightly less effort to implement (don't have to subtract 1). But I'm not too opinionated.

Comment (Asura5) [reply RepairsApi/V2/Boundary/Request/ListGasServiceDataRequest.cs:64]: The playbook is silent on this., so it's fine

File: RepairsApi/V2/Boundary/Response/ListGasServiceDataResponse.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+using System.Collections.Generic;
+
+namespace RepairsApi.V2.Boundary.Response
+{
+    public class ListGasServiceDataResponse
+    {
+        public int Page { get; set; }
```
Comment (Asura5) [inline RepairsApi/V2/Boundary/Response/ListGasServiceDataResponse.cs:7]: You should stick to the Hackney standard when choosing the pagination query parameter names. See https://playbook.hackney.gov.uk/API-Playbook/naming_conventions#must-stick-to-conventional-query-parameters

Comment (LBHCallumM) [reply RepairsApi/V2/Boundary/Response/ListGasServiceDataResponse.cs:7]: For page and pageSize, I used the same as MMH

**Review by Asura5 — APPROVED** [review]

**Pull request merged**